### PR TITLE
Adding Python 3.7 on Windows to test matrix

### PR DESCRIPTION
**Pull request by pierreluctg**
This is a **work in progress** since some of the unit test may have to be fixed or skipped on Windows.

If @vmalloc approve adding this test, I will push the other changes required in the PR. If not approved, I will create another PR for the failing unit tests in Windows.

### CODE REVIEW

**Comment by vmalloc** [comment]
I have no objection to add Windows support as long as it's not flaky (every other build breaking occasionally/sporadically).

**Comment by pierreluctg** [comment]
@vmalloc this is ready to go from my side.

**Comment by pierreluctg** [comment]
@vmalloc I have added Windows to the Travis allowed failures to address your concern about tests on Windows flakiness. I have not observed any issue, but keeping it there for now until there is enough Travis build to show.

File: tests/test_parallel.py [hunk 1]
```diff
@@ -12,6 +14,9 @@
 import tempfile
 import pytest
 
+if sys.platform.startswith("win"):
+    pytest.skip("does not run on windows", allow_module_level=True)
```
Comment (vmalloc) [inline tests/test_parallel.py]: Nice trick! Did not know that one :-)

**Pull request merged**